File: EazyCore/app/src/main/java/vn/eazy/sample/fragment/MainMenuFragment.java
package vn.eazy.sample.fragment;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.AppCompatButton;
import android.view.View;

import butterknife.BindView;
import butterknife.OnClick;
import vn.eazy.core.base.fragment.BaseMainFragment;
import vn.eazy.sample.R;

/**
 * Created by Harry on 4/16/17.
 */

public class MainMenuFragment extends BaseMainFragment {
    @BindView(R.id.btnMultiStateView)
    AppCompatButton btnMultiStateView;

    @BindView(R.id.btnNormalList)
    AppCompatButton btnNormalList;

    public static MainMenuFragment newInstance() {
        Bundle args = new Bundle();

        MainMenuFragment fragment = new MainMenuFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public int getLayoutId() {
        return R.layout.fragment_main_menu;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
    }

    @OnClick(R.id.btnMultiStateView)
    public void onMultiStateViewClick() {
        getFragmentHelper().replaceFragment(MultiStateViewFragment.newInstance());
    }

    @OnClick(R.id.btnNormalList)
    public void onNormalListClick() {
        getFragmentHelper().replaceFragment(NormalDataFragment.Companion.newInstance());
    }

}

File: EazyCore/app/src/main/java/vn/eazy/sample/fragment/MainMenuFragment.kt
package vn.eazy.sample.fragment

import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.fragment_main_menu.*
import vn.eazy.core.base.fragment.BaseMainFragment
import vn.eazy.sample.R

/**
 * Created by Harry on 4/16/17.
 */

class MainMenuFragment : BaseMainFragment(),View.OnClickListener{
    override fun getLayoutId(): Int {
        return R.layout.fragment_main_menu
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        btnMultiStateView.setOnClickListener(this)
        btnNormalList.setOnClickListener(this)
    }

    companion object {
        fun newInstance(): MainMenuFragment {
            val args = Bundle()
            val fragment = MainMenuFragment()
            fragment.arguments = args
            return fragment
        }
    }

    override fun onClick(v: View?) {
        when (v?.id){
            R.id.btnMultiStateView -> fragmentHelper.replaceFragment(MultiStateViewFragment.newInstance())
            R.id.btnNormalList -> fragmentHelper.replaceFragment(NormalDataFragment.newInstance())
        }
    }
}
